### ISSUE TITLE
Error in installing Openshift Origin 3.7, DNSMasq Configuration

### ISSUE THREAD

**Issue by ghost** [openshift/openshift-ansible#6622]
####  Installation Steps:

1.  Installed DNSMASQ:
     cat /etc/dnsmasq.d/*
     strict-order
     domain-needed
     local=/master-server.com/
     bind-dynamic
     log-queries
    address=/.master-server.com/192.168.68.73
------------------------------------------------------------------

2.  Networking Configuration look like this:
     cat /etc/sysconfig/network-scripts/ifcfg-eth0
     DEVICE="eth0"
     BOOTPROTO="dhcp" 
     BOOTPROTOv6="dhcp"
    ONBOOT="yes"
    TYPE="Ethernet"
    USERCTL="yes"
    PEERDNS="no"
    IPV6INIT="yes"
    PERSISTENT_DHCLIENT="1"
-----------------------------------------------------------------------
#### Using Advanced Installation 
git status
HEAD detached at openshift-ansible-3.7.19-1
------------------------------------------------------------------------
## Inventory File
[OSEv3:children]
masters
nodes
etcd

[OSEv3:vars]
ansible_ssh_user=balanna
ansible_become=true
containerized=false
deployment_type=origin
openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]
openshift_master_htpasswd_users={'admin': '$apr1$zgSjCrLt$1KSuj66CggeWSv.D.BXOA1', 'user': '$apr1$.gw8w9i1$ln9bfTRiD6OwuNTG5LvW50'}
openshift_release=v3.7.0
openshift_docker_insecure_registries=172.30.0.0/16
openshift_master_default_subdomain=master-server.com
openshift_use_dnsmasq=true

openshift_clock_enabled=true
openshift_master_cluster_method=native
openshift_registry_selector='registry=true'
openshift_router_selector='router=true'
openshift_hosted_router_selector='router=true'
openshift_master_cluster_public_hostname=master-server.com
openshift_public_hostname=master-server.com
openshift_disable_check=memory_availability,disk_availability

openshift_metrics_install_metrics=true
openshift_metrics_image_version=v3.7.0
openshift_metrics_project=openshift-infra
openshift_metrics_hawkular_hostname=master-server.com

[masters]
192.168.68.73    openshift_public_hostname=master-server.com openshift_hostname=master-server.com

[etcd]
192.168.68.73
[nodes]
192.168.68.73    openshift_hostname=master-server.com          openshift_node_labels="{'router':'true','registry':'true'}"

----------------------------------------------------------------------------------------
Once I ran the playbook, i am getting below error.

I am getting below errors:
" Could not resolve host:" error. see the below error

INSTALLER STATUS ******************************************************************************************************************************************
Initialization             : Complete
Health Check               : Complete
etcd Install               : Complete
Master Install             : Complete
Master Additional Install  : Complete
Node Install               : In Progress
        This phase can be restarted by running: playbooks/byo/openshift-node/config.yml

Failure summary:
  1. Hosts:    192.168.68.73
     Play:     Configure nodes
     Task:     Install Node package
     Message:  There was an error communicating with RHN.
               RHN channel support will be disabled.
               Error communicating with server. The message was:
               Name or service not known
               http://mirror.centos.org/centos/7/paas/x86_64/openshift-origin37/repodata/repomd.xml: [Errno 14] curl#6 - "Could not resolve host: mirror.centos.org; Name or service not known"
               Trying other mirror.
               http://mirror.centos.org/centos/7/paas/x86_64/openshift-origin37/origin-node-3.7.0-1.0.7ed6862.x86_64.rpm: [Errno 14] curl#6 - "Could not resolve host: mirror.centos.org; Name or service not known"
               Trying other mirror.
               http://mirror.centos.org/centos/7/paas/x86_64/openshift-origin37/tuned-profiles-origin-node-3.7.0-1.0.7ed6862.x86_64.rpm: [Errno 14] curl#6 - "Could not resolve host: mirror.centos.org; Name or service not known"
               Trying other mirror.

               Error downloading packages:
                 tuned-profiles-origin-node-3.7.0-1.0.7ed6862.x86_64: [Errno 256] No more mirrors to try.
                 origin-node-3.7.0-1.0.7ed6862.x86_64: [Errno 256] No more mirrors to try.

----------------------------------------------------------------------------------------
As DNSMasq is installed and configured, it is not resolving the hosts.
After adding the DNS nameserver in /etc/resolv.conf it start installing  

then i got this error.
Node Install               : In Progress
        This phase can be restarted by running: playbooks/byo/openshift-node/config.yml
Failure summary:

  1. Hosts:    192.168.68.73
     Play:     Configure nodes
     Task:     restart node
     Message:  Unable to restart service origin-node: Job for origin-node.service failed because the control process exited with error code. See "systemctl status origin-node.service" and "journalctl -xe" for details.

systemctl restart origin-node.service
Job for origin-node.service failed because the control process exited with error code. See "systemctl status origin-node.service" and "journalctl -xe" for details.

journalctl -xe 
Jan 04 23:05:30 ohtwapida20z origin-node[10122]: F0104 23:05:30.918203   10122 start_node.go:159] could not start DNS, unable to read config file: open /etc/origin/node/resolv.conf: no such file or directory

Once I create /etc/origin/node/resolv.conf file, then it works fine.

finally I am getting this error:
curl -k https://apiserver.kube-service-catalog.svc/healthz
[+]ping ok
[+]poststarthook/generic-apiserver-start-informers ok
[+]poststarthook/start-service-catalog-apiserver-informers ok
[-]etcd failed: reason withheld
healthz check failed

systemctl status etcd -l
● etcd.service - Etcd Server
   Loaded: loaded (/usr/lib/systemd/system/etcd.service; enabled; vendor preset: disabled)
   Active: active (running) since Thu 2018-01-04 23:56:56 UTC; 16s ago
 Main PID: 23247 (etcd)
   Memory: 56.6M
   CGroup: /system.slice/etcd.service
           └─23247 /usr/bin/etcd --name=master-server.com --data-dir=/var/lib/etcd/ --listen-client-urls=https://192.168.68.73:2379

Jan 04 23:56:56 ohtwapida20z etcd[23247]: 5fbc308a5215819 is starting a new election at term 2
Jan 04 23:56:56 ohtwapida20z etcd[23247]: 5fbc308a5215819 became candidate at term 3
Jan 04 23:56:56 ohtwapida20z etcd[23247]: 5fbc308a5215819 received MsgVoteResp from 5fbc308a5215819 at term 3
Jan 04 23:56:56 ohtwapida20z etcd[23247]: 5fbc308a5215819 became leader at term 3
Jan 04 23:56:56 ohtwapida20z etcd[23247]: raft.node: 5fbc308a5215819 elected leader 5fbc308a5215819 at term 3
Jan 04 23:56:56 ohtwapida20z etcd[23247]: published {Name:master-server.com ClientURLs:[https://192.168.68.73:2379]} to cluster be421405f663bdc3
Jan 04 23:56:56 ohtwapida20z etcd[23247]: ready to serve client requests
Jan 04 23:56:56 ohtwapida20z etcd[23247]: serving client requests on 192.168.68.73:2379
Jan 04 23:56:56 ohtwapida20z systemd[1]: Started Etcd Server.
Jan 04 23:56:56 ohtwapida20z etcd[23247]: Failed to dial 192.168.68.73:2379: connection error: desc = "transport: remote error: tls: bad certificate"; please retry.

curl -k https://192.168.68.73:2379
curl: (58) NSS: client certificate not found (nickname not specified)

**Comment by rackergs** [comment]
What do you get when you run

```
openssl s_client -connect 192.168.68.73:2379
```

**Comment by naveenbdevops** [comment]
Hi Rackergs, 

openssl s_client -connect 192.168.68.73:2379
CONNECTED(00000003)
depth=0 CN = ohtwapida20z.nss.vzwnet.com
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 CN = ohtwapida20z.nss.vzwnet.com
verify error:num=21:unable to verify the first certificate
verify return:1

**Comment by gregswift** [comment]
So I dont know what `ohtwapida20z.nss.vzwnet.com` is but it seems like my guess was right. Can you create a dns entry for 192.168.68.73 and start from the beginning?

**Comment by naveenbdevops** [comment]
Sure Greg, thanks.
is my dnsmasq configuration locking good
vim /etc/dnsmasq.conf
    resolv-file=/etc/resolv.dnsmasq
    host-record=ohtwapida20z.nss.vzwnet.com,192.168.68.73 
    # Wildcard DNS for OpenShift Applications - Points to Router
    address=/ohtwapida20z.nss.vzwnet.com/192.168.68.73 
    # Forward .local queries to SkyDNS
    server=/local/127.0.0.1#8053
    # Forward reverse queries for service network to SkyDNS.
    # This is for default OpenShift SDN - change as needed.
    server=/17.30.172.in-addr.arpa/127.0.0.1#8053
   
    cat /etc/dnsmasq.d/*
    strict-order
    domain-needed
     local=/ohtwapida20z.nss.vzwnet.com/
     bind-dynamic
     log-queries
     address=/.ohtwapida20z.nss.vzwnet.com/192.168.68.73

**Comment by gregswift** [comment]
i dont know enough about dnsmasq but as long as that syntax is right that looks reasonable

**Comment by naveenbdevops** [comment]
Hi Greg, 

I am getting this error.

systemctl restart origin-node.service
Job for origin-node.service failed because the control process exited with error code. See "systemctl status origin-node.service" and "journalctl -xe" for details.

journalctl -xe
Jan 08 13:05:30 ohtwapida20z origin-node[10122]: F0104 23:05:30.918203 10122 start_node.go:159] could not start DNS, unable to read config file: open /etc/origin/node/resolv.conf: no such file or directory

**Comment by rackergs** [comment]
I'm not sure whats going on with teh install not setting up a good verion. Have you re-reviewed the DNS prereqs? https://docs.openshift.com/container-platform/3.7/install_config/install/prerequisites.html#prereq-dns

**Comment by dsmithfv** [comment]
I just solved a similar issue.  The scaleup.yml was failing right after dnsmasqd started and resolv.conf was updated.

The solution was to lock down the dnsmasq version that we used on our original nodes.

yum install yum-plugin-versionlock

and put dnsmasq-2.76-2.el7_4.2.x86_64 in your:

/etc/yum/pluginconf.d/versionlock.list

The latest version 2.76-5 is breaking the Openshift installer on RHEL 7.4.
